### PR TITLE
feat: Create subcommandPermCheck plugin

### PR DESCRIPTION
**Pull request by Peter-MJ-Parker**
This plugin was requested to be uploaded for basic sub command permission handling. Only available for slash commands and will fault out for other command types.

## Plugin Submission Checklist

Before submitting your plugin, please ensure that you have followed the specifications below:

- [x] Your plugin code includes a JSDoc block with `@plugin` at the beginning and `@end` at the end.
- [x] The order of plugin metadata within the JSDoc block follows the structure provided:
  1. `@plugin`
  2. `description`
  3. `@author` (you can have multiple authors in this format: `@author  @jacoobes [<@182326315813306368>]`)
  4. `@version` (with the version number)
  5. `@example` (a nice example of how to use your plugin)
  6. `@end`

this is an example:
```js
/** 
 * @plugin
 * filters autocomplete interaction that pass the criteria
 * @author @jacoobes [<@182326315813306368>]
 * @version 1.0.0
 * @example
 * ```ts
 * import { CommandType, commandModule } from "@sern/handler";
 * import { filterA } from '../plugins/filterA.js'
 * export default commandModule({
 *    type : CommandType.Slash,
 *    options: [
 *       {  
 *          autocomplete: true,
 *          command : {
 *             //only accept autocomplete interactions that include 'poo' in the text
 *             onEvent: [filterA(s => s.includes('poo'))],
 *             execute: (autocomplete) => {
 *                let data = [{ name: 'pooba', value: 'first' }, { name: 'pooga', value: 'second' }]
 *                autocomplete.respond(data) 
 *             }
 *          }
 *       }
 *    ],
 *    execute: (ctx, args) => {}
 * })
 * @end
 */


```

## Plugin Submission Details
[Enter data here] this is optional
## Additional Notes

_[Include any additional information or notes you'd like to provide regarding your plugin submission.]_

### CODE REVIEW

**Review by jacoobes — APPROVED** [review]

File: plugins/subcommandPermCheck.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+export const permsToString = (...perms: PermissionResolvable[]) => {
+  return new PermissionsBitField(perms)
+    .toArray()
+    .map((perm) => `\`${perm}\``)
+    .join(", ");
+};
+
+export function subcommandPermCheck(opts: Options) {
+  return CommandControlPlugin<CommandType.Slash>(async (ctx, [, args]) => {
```
Comment (jacoobes) [inline plugins/subcommandPermCheck.ts]: might be a good idea to not use `[,args]`. Will be removed in favor of SDT in version 4. consider using `ctx.options`

File: plugins/subcommandPermCheck.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+//@ts-nocheck
+/**
+ * @plugin
+ * Inspired by the plugin "requirePermission" created by Benzo-Fury & needhamgary, this plugin will set permissions for specific subcommands without manually writing it into the code.
+ *
+ * @author @Peter-MJ-Parker [<@371759410009341952>]
+ * @version 1.0
+ * @example
+ * ```ts
+ * import { subcommandPermCheck } from "../plugins/subcommandPerms";
```
Comment (jacoobes) [inline plugins/subcommandPermCheck.ts]:
```suggestion
 * import { subcommandPermCheck } from "../plugins/subcommandPerms.js";
```

File: plugins/subcommandPermCheck.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ *  execute: ({ interaction }) => {
+ * 		//your code here
+ *  }
+ * })
+ * ```
+ * @end
+ */
+
+/** Marked TODO's will be reconfigured with release of sern v4! **/
+import { type GuildMember, type PermissionResolvable, type TextChannel } from 'discord.js';
```
Comment (jacoobes) [inline plugins/subcommandPermCheck.ts]:
```suggestion
import type { GuildMember, PermissionResolvable, TextChannel } from 'discord.js';
```

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: plugins/subcommandPermCheck.ts (hunk 2 above)
Comment (Peter-MJ-Parker) [reply plugins/subcommandPermCheck.ts]: Corrected!

File: plugins/subcommandPermCheck.ts (hunk 3 above)
Comment (Peter-MJ-Parker) [reply plugins/subcommandPermCheck.ts]: Corrected!

File: plugins/subcommandPermCheck.ts (hunk 1 above)
Comment (Peter-MJ-Parker) [reply plugins/subcommandPermCheck.ts]: This was going to be fixed when v4 was released but I fixed it!

**Review by Peter-MJ-Parker — COMMENTED** [review]
Finished reviewing and corrected conflicts.

**Review by jacoobes — APPROVED** [review]

**Pull request merged**